### PR TITLE
network: remove unnecessary link->ifname from debug log statements

### PR DESCRIPTION
**Pull request by floppym**
Since 98b0299479a68ffd414888368907fc776a46b82a, we log the interface name automatically via log_link_debug().

Fixes: https://github.com/systemd/systemd/issues/14782

### CODE REVIEW

**Comment by cdown** [comment]
Sure, removing it fixes the problem, but what was the intent of having it there in the first place, then? It does seem like useful information if it's actually populated at the time, is it really always NULL?

**Comment by floppym** [comment]
@cdown The information is redundant since 98b0299479a68ffd414888368907fc776a46b82a appends the interface name to the log message already.

**Comment by cdown** [comment]
Sure, that sounds fine. It might be worth documenting that in the commit message :-)

**Comment by floppym** [comment]
Commit message updated.

**Comment by cdown** [comment]
Looks ok to me, but I will leave it to network folks to say whether it's ok or not to only have this in metadata and not the message.

**Comment by floppym** [comment]
> It does seem like useful information if it's actually populated at the time, is it really always NULL?

I think the compiler cannot determine if it is never null, so it assumes the worst.

An alternative would be to wrap `link->ifname` with `strnull()` to ensure we never pass a `NULL` value.

**Comment by cdown** [comment]
> I think the compiler cannot determine if it is never null, so it assumes the worst.

That's what I thought, but we definitely have that in plenty of other places in our codebase (eg. format_bytes) where this doesn't happen, so I don't know what's really going on.

File: src/network/networkd-dhcp-server.c [hunk 1]
```diff
@@ -45,7 +45,7 @@ static int link_push_uplink_dns_to_dhcp_server(Link *link, sd_dhcp_server *s) {
         size_t n_addresses = 0, n_allocated = 0;
         unsigned i;
 
-        log_link_debug(link, "Copying DNS server information from %s", link->ifname);
+        log_link_debug(link, "Copying DNS server information");
```
Comment (keszybz) [inline src/network/networkd-dhcp-server.c]: Hmm, this will be rather misleading. When I see "hub0: Copying DNS server information from hub0", the meaning is pretty easy. But without the last two words, I would assume that the copy happens in the other direction. So maybe reword the message somehow, e.g. "Copying DNS server information from link"?

Comment (floppym) [reply src/network/networkd-dhcp-server.c]: Updated the messages as suggested.

**Comment by anitazha** [comment]
> That's what I thought, but we definitely have that in plenty of other places in our codebase (eg. format_bytes) where this doesn't happen, so I don't know what's really going on.

Not that it matters at this point but these may be occurring more recently because of https://github.com/systemd/systemd/issues/14691 or some new gcc thing.
